hazelcast/src/main/java/com/hazelcast/query/impl/Indexes.java

```diff
@@ -241,6 +249,25 @@ public InternalIndex[] getIndexes() {
         return indexes;
     }
 
+    public List<StepAwareStorage> getStepAwareStorages() {
+        List<StepAwareStorage> stepAwareStorages = new ArrayList<>();
+        for (InternalIndex index : indexes) {
+            List<StepAwareStorage> saStorages = index.getStepAwareStorages();
+            if (saStorages != null) {
+                stepAwareStorages.addAll(saStorages);
+            }
+        }
+        return stepAwareStorages;
+    }
+
+    public boolean isBackedByTieredStore() {
```

Comment:
is this method really needed? in the end we call `getStepAwareStorages`, it can just return an empty list if not `isBackedByTieredStore`.

Reply:
Addressed in https://github.com/hazelcast/hazelcast-enterprise/pull/6348/commits/1ac699f3a809f14817ffa665cd426b17d5b59320